packages/@expo/cli/src/start/interface/interactiveActions.ts

```diff
@@ -20,10 +20,13 @@ export class DevServerManagerActions {
     // If native dev server is running, print its URL.
     if (this.devServerManager.getNativeDevServerPort()) {
       try {
-        const url = this.devServerManager.getDefaultDevServer().getNativeRuntimeUrl()!;
+        const qrCodeUrl = this.devServerManager.getDefaultDevServer().getQRCodeUrl()!;
+        // even if the QR code points to the interstitial page, we want to display the native
+        // runtime URL
+        const nativeRuntimeUrl = this.devServerManager.getDefaultDevServer().getNativeRuntimeUrl()!;
 
-        printQRCode(url);
-        Log.log(printItem(chalk`Metro waiting on {underline ${url}}`));
+        printQRCode(qrCodeUrl);
+        Log.log(printItem(chalk`Metro waiting on {underline ${nativeRuntimeUrl}}`));
```

Comment:
The QR code is just a convenience feature, it should match the actual URL that is displayed. Showing a URL and a conditionally different QR code is confusing.

Reply:
I thought that showing a message saying `Metro waiting on <interstitial-page-url>` would be both confusing and inaccurate. I’ve updated this to print both URLs along with another message when the QR code points to the interstitial page.